Add notes/known issues section to installation page

joelhans opened the pull request:
<!--
Describe the change in summary section, including rationale and design decisions.
Include "Fixes #nnn" if you are fixing an existing issue.

In "Component Name" section write which component is changed in this PR. This
will help us review your PR quicker.

In "Test Plan" provide enough detail on how you plan to test this PR so that a reviewer can validate your tests. If our CI covers sufficient tests, then state which tests cover the change.

If you have more information you want to add, write them in "Additional
Information" section. This is usually used to help others understand your
motivation behind this change. A step-by-step reproduction of the problem is
helpful if there is no related issue.
-->

##### Summary

Fixes #8306 

I added this notes section to the installation page where we can put notices like SSL and workarounds for various distros. I don't love cluttering up that page any more, but I can't figure out where else we could put this information with the best mix of discoverability and not being too in-your-face.

##### Component Name

packaging/

##### Test Plan

<!---
Provide enough detail so that your reviewer can understand which test-cases you
have covered, and recreate them if necessary. If sufficient tests are covered
by our CI, then state which tests cover the change.
-->

##### Additional Information

Ferroin commented:
We might want to mention here that the build also currently does not work with LibreSSL in place of OpenSSL, and also fails when using Clang in some configurations.

joelhans commented:
@Ferroin Done!

Ferroin commented on packaging/installer/README.md:
```diff
@@ -165,3 +165,23 @@ the community helps fix any bugs that might have been introduced in previous rel
+`kickstart.sh` script on systems that have both a distribution-installed version of OpenSSL and a manually-installed
+local version. The Agent's installer cannot handle both, and so you must remove one or the other to install the Agent.
+
+**LibreSSL**: The Agent installer is only compatible with OpenSSL. Critical functions do not work on systems with
+LibreSSL installed as a replacement for OpenSSL.
```
Actually, the build fails completely if you try to use LibreSSL instead of OpenSSL. They use the same naming for libraries and headers but aren't API compatible.

thiagoftsm replied:
@Ferroin I am working on a new PR with TLS right now, I will bring a fix for this on it.

joelhans replied:
@Ferroin Done.

@thiagoftsm We can always edit this text again when your PR is ready.

prologic commented:
> We might want to mention here that the build also currently does not work with LibreSSL in place of OpenSSL, and also fails when using Clang in some configurations.

There was a PR to fix this. I'm not sure this is true anymore. Let me find the PR.

prologic commented:
Issue tracking LibreSSL fix #8796

prologic commented:
We should hold of fon merging this until #8796 is resolved otherwise it makes us look a bit silly to say it doens't compile with LibreSSL one day but does the next. :)

thiagoftsm commented:
@prologic the PR #9068 will fix the problem when it is merged, this is the motive I closed #8796.

joelhans commented:
> We should hold of fon merging this until #8796 is resolved otherwise it makes us look a bit silly to say it doens't compile with LibreSSL one day but does the next. :)

Waiting is fine with me. You guys can ping me when this is ready for an update and I'll get on it.

joelhans commented:
An update on this one: @zack-shoylev and I need to add a few more workarounds for old systems into this PR, so I changed the title to better reflect what we're doing and we'll ask for reviews again when we've added all the notes.

Context is at netdata/marketing#240.

zack-shoylev commented:
Should we add @ktsaou's notes about running on ubuntu 14 from https://github.com/netdata/netdata/issues/8905 ? will have to add some security warnings (essentially you are disabling security so a hostile intermediary could steal data you are uploading such as metrics).

prologic commented:
Probably a good idea. Should also mention that using the Static install is a valid solution here (_not even a work-around either_).

joelhans commented:
@zack-shoylev Of course—that's been the plan all along. I'm hoping to get it done today.

zack-shoylev commented on packaging/installer/README.md:
````diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
+export CFLAGS="-DACLK_SSL_ALLOW_SELF_SIGNED"
+```
+
+Then install Netdata using your method of choice.
````
> Note that this will leave your Netdata Cloud connection vulnerable to man-in-the-middle attacks and is not recommended.

prologic replied:
I _think_ we need to review this very carefully...

prologic commented on claim/README.md:
```diff
@@ -186,6 +186,15 @@ might be having with the ACLK or claiming process.
+6/7, your Agent may not be able to connect to Cloud.
+
+You should either reinstall Netdata with a [static build](/packaging/installer/methods/kickstart-64.md), or see the
+[known installation issues](/packaging/installer/README.md#failure-to-claim-connect-to-cloud) for an OpenSSL workaround
+that comes at the cost of decreased security.
```
Nit: This is __NOT__ a "work-around" and AFAIk effectively removes __ALL_ security. Without host verification you have absolutely zero idea who/what you are connecting to.

prologic commented on packaging/installer/README.md:
```diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
+Cloud.
+
+We recommend using the [static build installation](/packaging/installer/methods/kickstart-64.md) to avoid this issue.
+You can, however, use the workaround below.
+
+#### OpenSSL workaround
```
Again Nit: Not a work-around.

zack-shoylev replied:
I'm fine with removing the work-around wording, it's good to make it clear.

Do note that users running on old machines they are most likely (hopefully) just using for testing Netdata won't care they are disabling security.

prologic commented on packaging/installer/README.md:
```diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
+
+#### OpenSSL workaround
+
+Instead of using the static build, you can also reconfigure OpenSSL to allow self-signed certificates and skip hostname
```
@amoss / @netdata/infra Please correct me if I'm wrong here, but I don't think "self-signed" certs are really relevant here. It is simply that we are disabling hostname verification.

prologic replied:
In fact I'm kind of curious; @amoss Do we require all 3 bit flags here to make this work on systems with an older openssl that do not have `X509_VERIFY_PARAM_set1_host` feature

```#!c
LCCSCF_ALLOW_SELFSIGNED | LCCSCF_SKIP_SERVER_CERT_HOSTNAME_CHECK | LCCSCF_ALLOW_INSECURE
```

Is it not enough to set `LCCSCF_SKIP_SERVER_CERT_HOSTNAME_CHECK` option?

knatsakis replied:
@prologic I concur with both you comments:

> I don't think "self-signed" certs are really relevant here. It is simply that we are disabling hostname verification.

￼I thought so too.

> Is it not enough to set LCCSCF_SKIP_SERVER_CERT_HOSTNAME_CHECK option?

My understanding is that it should be enough.

amoss replied:
I think that it would be enough.

prologic replied:
Then I would prefer we remove the wording around "self-signed" and "work-around" and state what's actually going on here.

joelhans replied:
The self-signed stuff was my bad—I misinterpreted what the option was doing.

prologic commented on packaging/installer/README.md:
````diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
+
+Then install Netdata using your method of choice.
+
+#### Ubuntu 14.04 LTS
+
+To use the `CFLAGS` workaround on Ubuntu 14.04 LTS, you must first install `libuv1-dev` via a PPA:
+
+```bash
+add-apt-repository ppa:acooks/libwebsockets6
+apt-get update
+apt-get install libuv1-dev
+```
+
+Then proceed with the CFLAGS workaround described above.
+
````
This is already handled by #8916 not sure we need to duplicate the code in our docuemtnation as we don't do this for similar reasons for handling other systems.

joelhans replied:
That PR is not merged yet. Is some other fix already present for Ubuntu 14.04?

prologic commented on packaging/installer/README.md:
```diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
+
+Then proceed with the CFLAGS workaround described above.
+
+### CentOS 6 and CentOS 8
+
+To install the Agent on certain CentOS and RHEL systems, you must enable non-default repositories, such as EPEL or
+PowerTools, to gather hard dependencies. See the [CentOS 6](/packaging/installer/methods/manual.md#centos-rehel-6-x) and
+[CentOS 8](/packaging/installer/methods/manual.md#centos-rehel-8-x) sections for more information.
```
Why is this here? The installer takes care of this, it even tells the user and prompts then (_optionally_).

joelhans replied:
This is here because of the linked sections ([6](https://learn.netdata.cloud/docs/agent/packaging/installer/methods/manual/#centos--rhel-6x) and [8](https://learn.netdata.cloud/docs/agent/packaging/installer/methods/manual/#centos--rhel-6x)), which were added some months ago to explain how to install on these distros, since users were running into issues.

If those sections are no longer needed, let me know and I'll remove them in a separate PR.

prologic replied:
I would just add a note to say that our installer automatically takes cares of this if you use our `kickstart.sh`.

prologic commented on packaging/installer/README.md:
```diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
+
+We've received reports from the community about issues with running the `kickstart.sh` script on systems that have both
+a distribution-installed version of OpenSSL and a manually-installed local version. The Agent's installer cannot handle
+both, and so you must remove one or the other to install the Agent.
```
Alternatively use the Static Netdata.

prologic commented on packaging/installer/README.md:
```diff
@@ -165,3 +165,63 @@ the community helps fix any bugs that might have been introduced in previous rel
+a distribution-installed version of OpenSSL and a manually-installed local version. The Agent's installer cannot handle
+both, and so you must remove one or the other to install the Agent.
+
+### Clang compiler on Linux
+
+Our current build process has some issues when using certain configurations of the `clang` C compiler on Linux. See [the
+section on `nonrepresentable section on output`
+errors](/packaging/installer/methods/manual.md#nonrepresentable-section-on-output-errors) for a workaround.
```
These are actively being fixed and will no longer be an issue very soon.

joelhans commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Help Zack with docs on running Netdata on old systems](https://trello.com/c/QLKDYINS/54-help-zack-with-docs-on-running-netdata-on-old-systems)

joelhans commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Help Zack with docs on running Netdata on old systems](https://trello.com/c/QLKDYINS/54-help-zack-with-docs-on-running-netdata-on-old-systems)

joelhans commented:
@prologic Given that I wrote this PR on Friday and #9198 was then merged over the weekend, @zack-shoylev and I will need to take some time to figure out how to redesign this entire PR to accommodate those changes. More soon.

prologic commented:
Thanks @joelhans :) Let me know if you need any help!

joelhans commented:
@prologic This is actually ready for another review when you have time. I chatted with @zack-shoylev off GitHub about what's in the PR right now, and while he's happy with it, the opinions of you and @amoss are what really count!

joelhans commented:
@zack-shoylev Can you hop in here and make any last requests or approve? I think merging is blocked, in part, because you requested changes. Thanks!

prologic commented:
You had two approvals. Merged.
